kerlescan/rbac_service_interface.py

```diff
@@ -18,3 +18,76 @@ def get_perms(application, service_auth_key, logger, request_metric, exception_m
     perms = [perm["permission"] for perm in rbac_result["data"]]
 
     return perms
+
+
+def get_rbac_filters(rbac_data):
+    """Gets RBAC filters from RBAC data"""
+    # possible values in "groups.id" key:
+    # None - no filter applied, all hosts allowed
+    # [] - no groups ids present, no hosts allowed
+    # [id1, id2, None] - group ids present, only hosts with id1, id2 or no group (None) allowed
+
+    # TODO: select all permisions needed
+    permission_list = [
+        "inventory:hosts:read",
+        "inventory:hosts:*",
+        "inventory:*:read",
+        "inventory:*:*",
+    ]
+
+    # default result is to allow no hosts
+    group_ids = []
+
+    def get_attfibute_filter_value(filter):
+        value = filter.get("value", None)
+        if value is None:
+            return value
+        elif isinstance(value, list):
+            return value
+        elif isinstance(value, str):
+            # split by comma, parses "null" as None
+            return [None if val == "null" else val for val in value.split(",")]
+        else:
+            raise TypeError
+
+    # get only relevant permission records
+    definitions = [
+        perm["resourceDefinitions"] for perm in rbac_data if perm["permission"] in permission_list
+    ]
+
+    # if definition contains []
+    # then we have empty resource definition for allowed permission thus we allow all hosts
+    if [] in definitions:
+        group_ids = None
+
+    # flatten definitions so we can work with all of them at once
+    definitions = [item for row in definitions for item in row]
+
+    if group_ids is not None:
+        # no definition means no matched permission
+        if definitions == []:
+            group_ids = []
+        else:
+            # get attributeFilters relevant for inventory groups
+            attribute_filters = [
+                definition.get("attributeFilter", {})
+                for definition in definitions
+                if definition.get("attributeFilter", {}).get("key") == "group.id"
+            ]
+
+            if attribute_filters == []:
+                group_ids = None
```

Comment:
This should be returned if at least one of the permissions does not have an attribute filter, or has attribute filters that are not relevant.